internal/provider/data_source_gitlab_project_tag.go

```diff
@@ -16,47 +16,16 @@ var _ = registerDataSource("gitlab_project_tag", func() *schema.Resource {
 **Upstream API**: [GitLab API docs](https://docs.gitlab.com/ee/api/tags.html)`,
 
 		ReadContext: dataSourceGitlabProjectTagRead,
-		Schema: map[string]*schema.Schema{
-			"name": {
-				Description: "The name of a tag.",
-				Type:        schema.TypeString,
-				Required:    true,
+		Schema: constructSchema(
+			gitlabProjectTagGetSchema(),
```

Comment:
```suggestion
			datasourceSchemaFromResourceSchema(gitlabProjectTagGetSchema()),
```

Reply:
this will also make it the `message` attribute `read-only` ...

Reply:
Exactly, I have wrapped this function into `datasourceSchemaFromResourceSchema`.